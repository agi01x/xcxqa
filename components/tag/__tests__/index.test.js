import React from 'react';
import { mount } from 'enzyme';
import Tag from '..';

describe('Tag', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should be closable', () => {
    const onClose = jest.fn();
    const wrapper = mount(
      <Tag closable onClose={onClose} />
    );
    expect(wrapper.find('.anticon-close').length).toBe(1);
    expect(wrapper.find('.ant-tag').length).toBe(1);
    wrapper.find('.anticon-close').simulate('click');
    expect(onClose).toBeCalled();
    jest.runAllTimers();
    wrapper.update();
    expect(wrapper.find('.ant-tag').length).toBe(0);
  });

  it('should not be closed when prevent default', () => {
    const onClose = (e) => {
      e.preventDefault();
    };
    const wrapper = mount(
      <Tag closable onClose={onClose} />
    );
    expect(wrapper.find('.anticon-close').length).toBe(1);
    expect(wrapper.find('.ant-tag').length).toBe(1);
    wrapper.find('.anticon-close').simulate('click');
    jest.runAllTimers();
    expect(wrapper.find('.ant-tag').length).toBe(1);
  });

  describe('visibility', () => {
    it('can be controlled by visible with visible as initial value', () => {
      const wrapper = mount(
        <Tag visible />
      );
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: false });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: true });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
    });

    it('can be controlled by visible with hidden as initial value', () => {
      const wrapper = mount(
        <Tag visible={false} />
      );
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: true });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: false });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
});
