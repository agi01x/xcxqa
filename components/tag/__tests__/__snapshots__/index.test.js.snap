// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag visibility can be controlled by visible with hidden as initial value 1`] = `<span />`;

exports[`Tag visibility can be controlled by visible with hidden as initial value 2`] = `
<div
  class="ant-tag"
  data-show="true"
/>
`;

exports[`Tag visibility can be controlled by visible with hidden as initial value 3`] = `<span />`;

exports[`Tag visibility can be controlled by visible with visible as initial value 1`] = `
<div
  class="ant-tag ant-tag-zoom-appear"
  data-show="true"
/>
`;

exports[`Tag visibility can be controlled by visible with visible as initial value 2`] = `<span />`;

exports[`Tag visibility can be controlled by visible with visible as initial value 3`] = `
<div
  class="ant-tag"
  data-show="true"
/>
`;
